### PR TITLE
Created get_response_rate_by_period endpoint

### PR DESCRIPTION
**Pull request by dyang5200**

### CODE REVIEW

File: backend/api/views/main.py [hunk 1]
```diff
@@ -79,6 +79,28 @@ def get_response_rates():
...
 
+'''
+function that is called when you visit /rates_per_period
+Parameters
+    year: year string with format YY
+    optional tract_id: 11-digit tract id string
+    optional state: two digit id string
+Either date or year is required.
+'''
+@main.route("/rates_per_period", methods=["GET"])
```
Comment (coconut750750) [inline backend/api/views/main.py]: Can you also change the other endpoint from `response_rates/` to `/rate`

**Comment by caiyingshen** [comment]
~closed cuz duplicate~ jk

File: frontend/src/utils/apiWrapper.js [hunk 2]
```diff
@@ -25,12 +25,36 @@ export const getResponseByTractID = tract_id => {
     });
 };
 
-export const getResponseRatesByDate = date => {
-  const requestString = `${BASE_URL}response_rates?date=${date}`;
+export const getResponseRatesByYear = year => {
+  const requestString = `${BASE_URL}rate?year=${year}`;
```
Comment (chyku) [inline frontend/src/utils/apiWrapper.js:29]: i don't think these do the same thing- aren't you supposed to change just `response_rates` to `rate`?

**Review by chyku — CHANGES_REQUESTED** [review]
confusion

File: backend/api/views/main.py [hunk 3]
```diff
@@ -101,6 +118,7 @@ def populate_db():
         responses = parse_census_data(file, date, date_initial, parse2000)
         parse2000 = False
         for r in responses:
+            if r.tract_id[:2] != "17": continue
```
Comment (coconut750750) [inline backend/api/views/main.py]: Can you remove this line?

File: frontend/src/utils/apiWrapper.js [hunk 4]
```diff
@@ -25,12 +25,36 @@ export const getResponseByTractID = tract_id => {
...
+export const getResponseByTractIDAndYear = (tract_id, year) => {
+  /**
+   * Given:
+   * tract id in database
+   * year
+   *
+   * Returns all tracts associated to that id upon success
+   * Returns GET_TRACT_DATA_FAIL upon failure
+   */
+  const requestString = `${BASE_URL}response_rates?tract_id=${tract_id}&year=${year}`;
```
Comment (coconut750750) [inline frontend/src/utils/apiWrapper.js]: Can you change this from `response_rates` to `rate`

**Review by coconut750750 — CHANGES_REQUESTED** [review]

File: backend/api/views/response_rates.py [hunk 5]
```diff
@@ -5,68 +5,71 @@
...
-    {"tract_id": string, "rate": float},
+    {"tract_id": string, "rates": {"date": rate, ...}},
     ...
     ]
-The returned rate is the response rate on the last collection day in the year
 '''
 def get_response_rates_by_year(year, tract_id=None, state=None):
     response_rates = {}
 
-    responses = get_census_responses(tract_id, state);
```
Comment (chyku) [inline backend/api/views/response_rates.py]: omg who originally wrote this

**Review by chyku — APPROVED** [review]
uwu

**Pull request merged**